frontend/src/lib/newQueryBuilder/queryBuilderMappers/__tests__/mapQueryDataFromApiInputs.ts

```diff
@@ -0,0 +1,495 @@
+					},
+				],
+				legend: '{{service_name}}-{{operation}}',
+				reduceTo: 'sum',
+			},
+		],
+		queryFormulas: [],
+	},
+	id: '64fcd7be-61d0-4f92-bbb2-1449b089f766',
+	queryType: 'builder',
+} as unknown) as Query;
+
+export const replaceVariables = {
+	builder: {
+		queryData: [
+			{
+				aggregateAttribute: {
+					dataType: 'float64',
+					id: 'signoz_calls_total--float64----true',
+					isColumn: true,
+					isJSON: false,
+					key: 'signoz_calls_total',
+					type: '',
+				},
+				aggregateOperator: 'sum_rate',
+				dataSource: 'metrics',
+				disabled: false,
+				expression: 'A',
+				filters: {
+					items: [
+						{
+							key: {
+								dataType: 'string',
+								isColumn: false,
+								isJSON: false,
+								key: 'deployment_environment',
+								type: 'tag',
+							},
+							op: 'in',
+							value: 'default',
+						},
+						{
+							key: {
+								dataType: 'string',
+								isColumn: false,
+								isJSON: false,
+								key: 'service_name',
+								type: 'tag',
+							},
+							op: 'in',
+							value: 'frontend',
+						},
+						{
+							key: {
+								dataType: 'string',
+								isColumn: false,
+								isJSON: false,
+								key: 'operation',
+								type: 'tag',
+							},
+							op: 'in',
+							value: 'HTTP GET /dispatch',
+						},
+					],
+					op: 'AND',
+				},
+				functions: [],
+				groupBy: [
+					{
+						dataType: 'string',
+						id: 'service_name--string--tag--false',
+						isColumn: false,
+						isJSON: false,
+						key: 'service_name',
+						type: 'tag',
+					},
+					{
+						dataType: 'string',
+						id: 'operation--string--tag--false',
+						isColumn: false,
+						isJSON: false,
+						key: 'operation',
+						type: 'tag',
+					},
+				],
+				having: [],
+				legend: '{{service_name}}-{{operation}}',
+				limit: null,
+				orderBy: [],
+				queryName: 'A',
+				reduceTo: 'sum',
+				spaceAggregation: 'sum',
+				stepInterval: 60,
+				timeAggregation: 'rate',
+			},
+		],
+		queryFormulas: [],
+	},
+	clickhouse_sql: [
+		{
+			disabled: false,
+			legend: '',
+			name: 'A',
+			query: '',
+		},
+	],
+	id: 'test-id',
+	promql: [
+		{
+			disabled: false,
+			legend: '',
+			name: 'A',
+			query: '',
+		},
+	],
+	queryType: 'builder',
+	unit: undefined,
+};
+
+export const defaultOutput = {
+	builder: {
+		queryData: [
+			{
+				ShiftBy: 0,
+				aggregateAttribute: {
+					dataType: 'float64',
+					isColumn: true,
+					isJSON: false,
+					key: 'system_disk_operations',
+					type: 'Sum',
+				},
+				aggregateOperator: 'rate',
+				dataSource: 'metrics',
+				disabled: false,
+				expression: 'A',
+				filters: { items: [], op: 'AND' },
+				functions: [],
+				groupBy: [],
+				having: [],
+				legend: '',
+				limit: 0,
+				offset: 0,
+				orderBy: [],
+				pageSize: 0,
+				queryName: 'A',
+				reduceTo: 'avg',
+				spaceAggregation: 'sum',
+				stepInterval: 240,
+				timeAggregation: 'rate',
+			},
+		],
+		queryFormulas: [],
```

Comment:
It is possible that the user can go to the alerts page from the dashboard and then make the changes on the alerts query builder. What happens when the user goes to the alerts page with `A/B` and then on the alerts page deletes `F1` and `B`, Will the step interval remain unchanged?

Reply:
once the user navigates from the dashboards page to the alerts page our task is to take the query and move the same as is to alerts with the variable values filled. post that user can do whatever changes they feel necessary in the alerts page. we do not block any operation there.

Reply:
added test cases for multiple queries with functions as well